### PR TITLE
Feature/issue 2961 change cache background

### LINKED ISSUE #2961: change_cache.go goroutine background tasks are hanging around longer than necessary
Factored out of https://github.com/couchbase/sync_gateway/issues/2938#issuecomment-334889877

Since the change cache background tasks are in a sleep loop (and in my case, sleeping for 1 minute), there is a long delay between singalling that they should stop and them actually stopping.

When running the unit tests in a suite, this is a resource leak.

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #2961

### CODE REVIEW

**Comment by tleyden** [comment]
Regarding:

> This test currently fails when running in integration mode since tearDownTestDB() doesn't close the database in that case

That will presumably be removed as part of https://github.com/couchbase/sync_gateway/issues/2938 (and has been already on the branch for that issue), but I held off doing in this PR to keep this PR as small as possible.

File: db/change_cache.go [hunk 1]
```diff
@@ -86,6 +87,7 @@ func (c *changeCache) Init(context *DatabaseContext, lastSequence SequenceID, on
 	c.onChange = onChange
 	c.channelCaches = make(map[string]*channelCache, 10)
 	c.receivedSeqs = make(map[uint64]struct{})
+	c.terminator = make(chan bool, 1) // give it buffer size of 1 so something trying to send won't block forever
```
Comment (adamcfraser) [inline db/change_cache.go]: We don't ever actually want to send to this channel, though, right?  Seems like giving it a buffer of 1 might result in future errors slipping through.

File: db/change_cache_test.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/couchbase/sync_gateway/channels"
 
 	"github.com/couchbaselabs/go.assert"
+	"runtime"
+	"regexp"
```
Comment (adamcfraser) [inline db/change_cache_test.go]: These imports need to be reorganized.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/change_cache.go (hunk 1 above)
Comment (tleyden) [reply db/change_cache.go]: Good catch.  Originally I was sending to the channel, then later signalled via closing the channel.  Will fix.

**Comment by tleyden** [comment]
@adamcfraser ok both PR feedback items are addressed.

Integration tests:

http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/307/
http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/308/

**Comment by tleyden** [comment]
Integration tests passed

**Pull request merged**